langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/GrpcAgentProcessor.java

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.agents.grpc;
+
+import ai.langstream.agents.grpc.ProcessorGrpc.ProcessorStub;
+import ai.langstream.api.runner.code.AbstractAgentCode;
+import ai.langstream.api.runner.code.AgentProcessor;
+import ai.langstream.api.runner.code.RecordSink;
+import ai.langstream.api.runner.code.SimpleRecord;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.protobuf.ByteString;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.avro.Conversions;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+
+@Slf4j
+public class GrpcAgentProcessor extends AbstractAgentCode implements AgentProcessor {
+
+    protected ManagedChannel channel;
+    private ProcessorStub processorStub;
+    private StreamObserver<ProcessorRequest> request;
+    private RecordSink sink;
+
+    // For each record sent, we increment the recordId
+    private final AtomicInteger recordId = new AtomicInteger(0);
```

Comment:
probably we need AtomicLong here, it is possible to reach 2 billion messages for a production system that is processing data

Reply:
Of course! I use long everywhere for record id (protocol, caches) except here... Good catch !

Reply:
Note that it wouldn't have been a real issue since record ids are transient and forgotten once we get the server response. I believe there would be an OOM before reaching the MAX_VALUE if they are not removed.

Reply:
done